File: web-client/src/model/ArticleInfo.ts
import { LOCALE } from '../types';

export interface ArticleInfoType {
  id: string
  articleListId: string
  title: string
  author: string
  createAt: string
  locale: string
  visible: boolean
}

export const FAKE_ID = '<ARTICLE_ID>';
export const FAKE_LIST_ID = '<ARTICLE_LIST_ID>';
class ArticleInfo {
  id: string;
  articleListId: string;
  title: string;
  author: string;
  createAt: Date;
  locale: LOCALE;
  visible: boolean;
  constructor (data: ArticleInfoType) {
    this.id = data.id;
    this.articleListId = data.articleListId;
    this.title = data.title;
    this.author = data.author;
    this.createAt = new Date(data.createAt);
    this.locale = LOCALE[data.locale as keyof typeof LOCALE];
    this.visible = data.visible;
  }
}

export default ArticleInfo;

File: src/main/java/com/utticus/blogloo/entity/ArticleInfo.java
package com.utticus.blogloo.entity;

import javax.persistence.Column;
import java.util.Date;
import java.util.UUID;

public class ArticleInfo {
    private UUID id;

    private UUID articleListId;

    private String locale;

    private String title;

    private String author;

    private Date createAt;

    public ArticleInfo() {}

    public ArticleInfo(UUID id, UUID articleListId, String locale, String title, String author, Date createAt) {
        this.id = id;
        this.articleListId = articleListId;
        this.locale = locale;
        this.title = title;
        this.author = author;
        this.createAt = createAt;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public UUID getArticleListId() {
        return articleListId;
    }

    public void setArticleListId(UUID articleListId) {
        this.articleListId = articleListId;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public Date getCreateAt() {
        return createAt;
    }

    public void setCreateAt(Date createAt) {
        this.createAt = createAt;
    }
}
